tests/dummy/app/pods/button/controller.js

```diff
@@ -1,9 +1,27 @@
 import Ember from 'ember'
 const {Controller} = Ember
 
+function createActionClosure (func) {
+  let context = this
+  return function closure () {
+    func.apply(context, arguments)
+  }
+}
+
 export default Controller.extend({
   vertical: false,
 
+  // BEGIN-SNIPPET button-hash
+  createAttrsHash: Ember.on('init', function () {
+    this.set('attrsHash', {
+      priority: 'primary',
+      size: 'medium',
+      text: 'Data driven',
+      onClick: createActionClosure.call(this, this.actions.onClickHandler)
+    })
```

Comment:
Can absolutely clean up the DX here, but the point is that the button's attrs were defined outside of the template, can include anything the button supports and could easily be data driven